ui/component/or-tree-menu/src/index.ts

```diff
@@ -0,0 +1,732 @@
+import {css, html, LitElement, PropertyValues, TemplateResult} from "lit";
+import {customElement, property, queryAll} from "lit/decorators.js";
+import {map} from "lit/directives/map.js";
+import {when} from "lit/directives/when.js";
+import {InputType} from "@openremote/or-mwc-components/or-mwc-input";
+import {getContentWithMenuTemplate} from "@openremote/or-mwc-components/or-mwc-menu";
+import {ListItem} from "@openremote/or-mwc-components/or-mwc-list";
+import {Util} from "@openremote/core";
+import {OrTreeNode} from "./or-tree-node";
+import {OrTreeGroup} from "./or-tree-group";
+import {moveNodesToGroupNode} from "./util";
+import {OrTreeDragEvent, OrTreeSelectEvent, TreeMenuSelection, TreeMenuSorting, TreeNode} from "./model";
+
+import "./or-tree-group";
+import "./or-tree-node";
+
+export * from "./or-tree-group";
+export * from "./or-tree-node";
+export * from "./model";
+
+const styles = css`
+    * {
+        box-sizing: border-box;
+    }
+    
+    #tree-container {
+        display: flex;
+        flex-direction: column;
+        height: 100%;
+    }
+
+    #tree-list {
+        flex: 1;
+        overflow: hidden auto;
+        list-style: none;
+        padding: 0;
+        margin: 0;
+    }
+
+    or-tree-node > or-tree-node > * {
+        pointer-events: none;
+    }
+    
+    or-tree-node {
+        transition: background-color 80ms;
+    }
+    
+    or-tree-node[drophover] {
+        background: #e9ecef;
+    }
+    
+    #tree-header {
+        display: flex;
+        justify-content: space-between;
+        align-items: center;
+        padding: 0 15px;
+        min-height: 48px;
+        background: var(--or-app-color4, #4d9d2a);
+        color: var(--or-app-color7, white);
+        --or-icon-fill: var(--or-app-color7, white);
+    }
+    
+    #tree-header-title {
+        margin: 0;
+        font-weight: 500;
+        font-size: 16px;
+    }
+    
+    #tree-header-actions {
+        display: flex;
+        align-items: center;
+    }
+`;
+
+/**
+ * @event {OrTreeSelectEvent} or-tree-select - Triggers upon selecting a node, and dispatches a list of the nodes selected.
+ * @event {OrTreeDragEvent} or-tree-drag - Triggers upon dragging a node to a new group, and dispatches a list of dragged nodes, the group node, and the updated list of all nodes.
+ */
```

Comment:
I see these warnings in IntelliJ:

```
 Unresolved variable or type 'event:OrTreeSelectEvent'
 Unresolved variable or type 'event:OrTreeDragEvent'
```

can that be prevented?

Reply:
I assume this occurs because of the inheritance on `CustomEvent` instead of the generic `Event` type.
This format is however used by the [custom-elements-manifest](https://custom-elements-manifest.open-wc.org) library for identifying component events.